### PR TITLE
Switch BCL testing from nunit to nunitlite

### PR DESCRIPTION
**Pull request by akoeplinger**

### CODE REVIEW

File: mcs/build/tests.make [hunk 1]
```diff
@@ -147,19 +97,36 @@ test-local-aot-compile: $(topdir)/build/deps/nunit-$(PROFILE).stamp
 
 endif # ALWAYS_AOT
 
+NUNITLITE_CONFIG_FILE=$(topdir)/class/lib/$(PROFILE)/nunit-lite-console.exe.config
```
Comment (marek-safar) [inline mcs/build/tests.make]: Change it to `$(topdir)/class/lib/$(PROFILE)/$(PARENT_PROFILE)nunit-lite-console.exe.config` and you could remove the ifdef bellow

File: mcs/class/System.Data.Linq/build/Makefile [hunk 2]
```diff
@@ -25,15 +25,14 @@ sqlite_tests_dep = \
 
 $(sqlite_tests): $(sqlite_tests_dep)
 	$(TEST_COMPILE) -target:library -out:$@ \
-		-r:System.Data.dll -r:$(the_assembly) -r:Mono.Data.Sqlite $(test_nunit_ref) \
+		-r:../../lib/net_4_x/System.Data.dll -r:$(the_assembly) -r:../../lib/net_4_x/Mono.Data.Sqlite.dll -r:../../lib/net_4_x/System.Core.dll -r:../../lib/net_4_x/System.dll -r:../../lib/net_4_x/System.Xml.dll $(test_nunit_ref) \
```
Comment (marek-safar) [inline mcs/class/System.Data.Linq/build/Makefile]: It should use `-r:$(topdir)/class/lib/$(PROFILE)/` for path reference

File: mcs/class/System.Web/Test/System.Web/HttpRuntimeTest.cs [hunk 3]
```diff
@@ -55,9 +55,9 @@ public static void UnloadAppDomain_OnLoad (Page p)
 		[Test]
 		[Category ("NunitWeb")]
 		[Ignore ("Pending fix for bug 351878")]
-		[Explicit]
 		public void UnloadAppDomain100Times () 
 		{
+			Assert.Ignore ("// TODO: this crashes when run under nunitlite after some time");
```
Comment (marek-safar) [inline mcs/class/System.Web/Test/System.Web/HttpRuntimeTest.cs]: Isn't this already ignored by using [Ignore] ?

File: mcs/class/test-helpers/NunitHelpers.cs [hunk 4]
```diff
@@ -0,0 +1,112 @@
+using System.IO;
+using System.Collections;
+using NUnit.Framework.Constraints;
+
+namespace NUnit.Framework
+{
+	class CollectionAssert
```
Comment (marek-safar) [inline mcs/class/test-helpers/NunitHelpers.cs]: Change all classes to be `static`

File: mcs/class/System.Web/Test/System.Web/HttpRuntimeTest.cs (hunk 3 above)
Comment (akoeplinger) [inline mcs/class/System.Web/Test/System.Web/HttpRuntimeTest.cs]: No, it was still getting run, I added more details in the commit message: https://github.com/akoeplinger/mono/commit/95f9f418e836d205af0b15dfdaccd3c2273aefcf

This was just some leftover debugging helper that I forgot to remove when squashing.

**Pull request merged**